restli-tools/src/main/java/com/linkedin/restli/tools/snapshot/gen/PegasusSchemaSnapshotExporter.java

```diff
@@ -0,0 +1,130 @@
+
+  private static final String PDL = "pdl";
+
+  private static final String PDSC = "pdsc";
+
+  /**
+   * Generate pegasus schema snapshot(pegasusSchemaSnapshot.pdl) files to the provided output directory
+   * based on the given input pegasus schemas.
+   *
+   * @param resolverPath schema resolver path
+   * @param inputPath input files directory
+   * @param outputDir output files directory
+   * @return GeneratorResult
+   * @throws IOException
+   */
+  public void export(String resolverPath, String inputPath, File outputDir) throws IOException
+  {
+    List<DataSchema> dataSchemas = parseDataSchema(resolverPath, inputPath);
+    for (DataSchema dataSchema : dataSchemas)
+    {
+      writeSnapshotFile(outputDir, ((NamedDataSchema) dataSchema).getFullName(), dataSchema);
+    }
+  }
+
+  private static List<DataSchema> parseDataSchema(String resolverPath, String inputPath)
+      throws RuntimeException, FileNotFoundException
+  {
+    Iterator<File> iterator = FileUtils.iterateFiles(new File(inputPath), new String[]{PDL, PDSC}, true);
+    List<DataSchema> schemas = new ArrayList<>();
+    while (iterator.hasNext())
+    {
+      DataSchemaResolver resolver = MultiFormatDataSchemaResolver.withBuiltinFormats(resolverPath);
+      File inputFile = iterator.next();
+      String fileExtension = getFileExtension(inputFile.getName());
+      PegasusSchemaParser parser = AbstractSchemaParser.parserForFileExtension(fileExtension, resolver);
+      parser.parse(new FileInputStream(inputFile));
+
+      if (parser.hasError())
+      {
+        throw new RuntimeException(parser.errorMessage());
+      }
+
+      List<DataSchema> topLevelDataSchemas = parser.topLevelDataSchemas();
+      if (topLevelDataSchemas == null || topLevelDataSchemas.isEmpty() || topLevelDataSchemas.size() > 1)
+      {
+        throw new RuntimeException("Could not parse schema : " + inputFile.getAbsolutePath());
+      }
+      DataSchema topLevelDataSchema = topLevelDataSchemas.get(0);
+      if (!(topLevelDataSchema instanceof NamedDataSchema))
+      {
+        throw new RuntimeException("Invalid schema : " + inputFile.getAbsolutePath() + ", the schema is not a named schema.");
+      }
+      schemas.add(topLevelDataSchema);
+    }
+    return schemas;
+  }
+
+  private static void writeSnapshotFile(File outputDir, String fileName, DataSchema dataSchema) throws IOException
+  {
+    StringWriter stringWriter = new StringWriter();
+    SchemaToPdlEncoder schemaToPdlEncoder = new SchemaToPdlEncoder(stringWriter);
+    schemaToPdlEncoder.setTypeReferenceFormat(AbstractSchemaEncoder.TypeReferenceFormat.DENORMALIZE);
+    schemaToPdlEncoder.encode(dataSchema);
+
+    File generatedSnapshotFile = new File(outputDir, fileName + PEGASUS_SCHEMA_SNAPSHOT_SUFFIX);
+
+    if (generatedSnapshotFile.exists())
+    {
+      if (!generatedSnapshotFile.delete())
+      {
+        throw new IOException(generatedSnapshotFile + ": Can't delete previous version");
+      }
+    }
```

Comment:
Is this necessary, given that the way `FileOutputStream` is constructed later is to overwrite any existing files?

Reply:
Good catch, removed.